DenisovPavel #Lab3

Ppasha9 opened the pull request.

Commit:
Add daemon

- "first" final version of daemon

Commit:
Fix pid file mistakes

- create pid file in 'build.sh' script
- do not remove pid file in daemon code

Commit:
Fix mistakes

- use 'openlog' function
- exchange defines WARNING, INFO and ERROR by directly calling 'syslog' function
- delete unused logging

Ppasha9 committed:
Add second lab

Ppasha9 committed:
Fix include header

- replace "bits/types/siginfo_t.h" by "csignal"

Ppasha9 committed:
Add errno.h

- add errno.h header in conn.h

Ppasha9 committed:
Hot fix #1

- remove unused compile parameter from makefile
- correct log messages in client and host

Ppasha9 committed:
Fix issues

- make const variables in host static
- delete global variables from connections' implementations
- now you should input new number in host every round
- add SIGINT singal support in host

Ppasha9 committed:
Cosmetic fixes

Ppasha9 committed:
Fix issues

- remove 'explicit' in client.h
- new order of host/client communication
- new support of additional arguments in connections' implementation
- add hack for socket connection implementation, in case when client failed and new client is attached

Ppasha9 committed:
Hot fix

Commit:
Fix waiting client section in host

Ppasha9 committed:
Add Lab3 implementation

KotsMikhail reviewed (changes requested):
Общий тест для множества с ленивой синхронизацией иногда зависает. Запускал с дефолтным конфигом.
Так же форматирование фигурных скобок разное в разных местах

Commit:
Fix issues

- delete printing the name of config file
- correct brackets
- add "false" return value in some methods

Ppasha9 commented:
Поправил скобки в одном месте.
Возможно Вы имели в виду, что фигурные скобки в случае функции и методов расположены не так, как в случае с конструкциями типа if, while, etc.
Я просто привык к такому стилю кодирования, так как пишу уже на нем пару лет.
И также считаю, что в этом ничего плохого, так как для каждого типа конструкций форматирование скобок остается одинаковым во всей программе.
То есть для функций и методов форматирование одно, а для конструкций if, while другое. По-моему это ничему не противоречит.

Также убрал вывод полного пути к конфигурационному файлу, эта было для дебага.

Ppasha9 commented:
Насчет зависания общего теста для множества с ленивой синхронизацией.
Странно, что он у Вас зависает. Я тестировал на двух разных устройствах, и на обоих все было нормально. Особенно при дефолтном конфиге. В некоторых случаях, если указать большое количество элементов и потоков, то придется подождать некоторое количество времени, но ничего не зависало.
Для того, чтобы удостовериться, я воспользовался утилитой `valgrind`, которая доступна по умолчанию почти во всех дистрибутивах Linux, ну или ее легко можно установить через `sudo apt install valgrind`.
Для теста было несколько раз запущено программа, выполняющая 100 раз подряд общий тест для множества с ленивой синхронизацией. Ничего не зависло. На скриншоте можно посмотреть что мне выдала утилита.
![image](https://user-images.githubusercontent.com/12701184/71243661-dc970880-2321-11ea-997c-6a44849275df.png)
Время такое большое, потому что при работе программы утилита проверяет утечки памяти, segmentation fault и так далее.
Если запускать без `valgrind`, то все выполняется еще быстрее.

SWAGLORD1 commented:
Вы можете отправить скрин с выводом результатов тестов для 1000 записей и 100 потоков?
У меня общий тест зависает на большом количестве потоков, для 10 все нормально, по коду не могу понять, в чем дело. Вы на виртуалке это запускаете?

Ppasha9 commented:
Прикладываю скриншоты, на которых показаны параметры дефолтного файла и выход.
На втором скриншоте - замер времени выполнения одного общего теста для множества с ленивой синхронизацией.
На третьем - вывод всей программы.
Да, я запускаю на виртуалке.
![image](https://user-images.githubusercontent.com/12701184/71313582-c71dfd80-244b-11ea-803e-6fe77da56f29.png)
![image](https://user-images.githubusercontent.com/12701184/71313586-d3a25600-244b-11ea-81f2-26fc02744775.png)
![image](https://user-images.githubusercontent.com/12701184/71313588-de5ceb00-244b-11ea-934d-6dbbbbeb1b73.png)

Ppasha9 commented:
Извиняюсь, что закрыл пулл реквест.
Кнопки для комментирования и закрытия реквеста рядом находятся....

Ppasha9 commented:
Запустил программу с отдельным замером времени для теста читателей, теста писателей и общего теста. Использовал конфигурационный файл, скриншот которого указан в прошлом комментарии.
![image](https://user-images.githubusercontent.com/12701184/71320452-90340000-24bc-11ea-82e7-89eaa0bb7d12.png)

SWAGLORD1 commented:
Засчитано
